WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/RefundShippingLabelViewController.swift

```diff
@@ -0,0 +1,204 @@
+import UIKit
+import Yosemite
+
+/// Displays information about the refund with a CTA to request a refund.
+final class RefundShippingLabelViewController: UIViewController {
+    @IBOutlet private weak var tableView: UITableView!
+    @IBOutlet private weak var refundButton: UIButton!
+
+    private let shippingLabel: ShippingLabel
+    private let viewModel: RefundShippingLabelViewModel
+    private let rows: [Row]
+    private let noticePresenter: NoticePresenter
+    private let onDismiss: () -> Void
+
+    init(shippingLabel: ShippingLabel,
+         currencyFormatter: CurrencyFormatter = .init(currencySettings: ServiceLocator.currencySettings),
+         noticePresenter: NoticePresenter = ServiceLocator.noticePresenter,
+         onDismiss: @escaping () -> Void) {
+        self.shippingLabel = shippingLabel
+        self.viewModel = RefundShippingLabelViewModel(shippingLabel: shippingLabel, currencyFormatter: currencyFormatter)
+        self.rows = [.purchaseDate, .refundableAmount]
+        self.noticePresenter = noticePresenter
+        self.onDismiss = onDismiss
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        configureNavigationBar()
+        configureTableView()
+        configureRefundButton()
+    }
+}
+
+// MARK: Action Handling
+private extension RefundShippingLabelViewController {
+    func refundShippingLabel() {
+        let action = ShippingLabelAction.refundShippingLabel(shippingLabel: shippingLabel) { result in
+            self.showRefundResultNotice(result: result)
+        }
+        ServiceLocator.stores.dispatch(action)
+        onDismiss()
```

Comment:
What do you think about moving this to the `ViewModel` layer?
That way we would have a more clear input and output action that will make it easy to unit test. 
Also, we could inject `ServiceLocator.stores` and prevent any kind of "shared state craziness" in tests.

Reply:
yea sounds like a good idea, updated in https://github.com/woocommerce/woocommerce-ios/pull/3306/commits/a4d9c1f7c05a73b0a134a39dec5f545b8af1c38d.

originally I imagined the view model to just provide calculated UI properties for view display, but including the action in the view model does help with testing 😃